concatenate.py

```diff
@@ -0,0 +1,71 @@
+import argparse
+import sys
+from os.path import exists
+
+import pandas as pd
+
+parser = argparse.ArgumentParser(
+    description="Concatenate multiple transactions spreadsheets into one"
+)
+parser.add_argument(
+    "spreadsheets",
+    type=str,
+    nargs="+",
+    help="Spreadsheets to concatenate",
+)
+parser.add_argument(
+    "-o",
+    "--output",
+    dest="output",
+    type=str,
+    required=True,
+    help="Concatenated .xlsx file",
+)
+
+
+def main(arguments):
+    merged = {}
+    transaction_num = {}
+    for file in arguments.spreadsheets:
+        print(f"Processing '{file}'")
+        if not exists(file):
+            print(f"Spreadsheet '{file}' doesn't exist.", file=sys.stderr)
+            exit(1)
+
+        spreadsheets = pd.read_excel(file, sheet_name=None)
+        for name, sheet in spreadsheets.items():
+            if name not in merged:
+                merged[name] = [sheet]
+                transaction_num[name] = len(sheet)
+            elif not sheet.empty:
+                transaction_num[name] += len(sheet)
+                # Do not let to have empty values - pd.concat cannot handle it anymore
+                if merged[name][-1].empty:
+                    merged[name][-1] = sheet
+                else:
+                    merged[name].append(sheet)
+
+    print("Concatenating files and writing out the result")
```

Comment:
Could you add a simple logic here to check for exact duplicates of transactions, e.g. when someone exported reports with overlapping export dates and/or a logic to remove exact duplicates?

Reply:
Yes, I can try to add it here. Probably sometimes next week.